### ISSUE TITLE
Let formatjs handle language-specific external links

### ISSUE THREAD

**Issue by patcon** [GCTC-NTGC/gc-digital-talent#3269]
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description
Sometimes we have external links that have different paths for English and French. Right now, we handle these localized links with either (a) our own logic, or (b) only an english link:

(a) own logic
https://github.com/GCTC-NTGC/gc-digital-talent/blob/b972691d1dafaebf8bdcaa19ba477b0279f517f6/frontend/talentsearch/src/js/components/languageInformationForm/LanguageInformationForm.tsx#L121-L128

(b) only english
https://github.com/GCTC-NTGC/gc-digital-talent/blob/b972691d1dafaebf8bdcaa19ba477b0279f517f6/frontend/talentsearch/src/js/components/roleSalaryForm/RoleSalaryForm.tsx#L263-L268

Since arbitrary components/elements can be swapped in via rich text, we could have an arbitrary `<LocalizedLink />` component that "double localizes" in that the `href` attribute also localizes. These urls would be translated with the rest of the strings. I imagine it could look something like this, and `<LocalizedLink />` would capture the logic in a component for wherever we need it :)

```jsx
const LocalizedLink = ({ href, i18nDescription, children, ...rest) => {
  const { formatMessage } = useIntl();
  return (
    <a
      href={ formatMessage({
        defaultMessage: href,
        description: i18nDescription
       })}
      {...rest}
    >{ children }</a>
  )
}

const MyComponent = () => {
  const { formatMessage } = useIntl();
  return (
    <div>
      { formatMessage({
        defaultMessage: "Click <link>here</link> for an English link",
        values: {
          link: chunks => (
            <LocalizedLink href="http://example.com/en" i18nDescription="The link in MyComponent">
              {chunks}
            </LocalizedLink>
          )
        },
      }) }
    </div>
  )
}
```

Resources:
- https://stackoverflow.com/questions/41129628/how-can-i-formatmessage-with-a-tags-links-using-react-intl
- https://formatjs.io/docs/react-intl/components#function-as-the-child

## Acceptance Criteria
- [ ] ...

   [image]: https://user-images.githubusercontent.com/305339/166570935-6425a615-82c7-4b2f-9840-5c7d698fa2ab.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

**Comment by patcon** [comment]
Hm. Maybe the code that injects autogenerated IDs doesn't work on dynamic values :/  I'm realizing the examples all use an `id`, but I didn't realize that was important to it working

I know we don't use manual IDs, but does keying the LocalizedLink with an ID prop (and passing that in) allow this to work? Or does using defineMessages rather than formatMessage prevent the issue?

It's possible this won't work though, so I guess if it insists on an ID and can't autogen, then it might be a losing battle 😞

**Comment by esizer** [comment]
> I know we don't use manual IDs, but does keying the LocalizedLink with an ID prop (and passing that in) allow this to work?

I can try but, that means we cannot use description since `react-intl` does not allow both an `id` and `description` which could cause issues if we do not make the `id` descriptive enough. I know the examples have both but when i try it gets automatically removed and flagged as an error in `eslint` 🤷

**Comment by patcon** [comment]
> `react-intl` does not allow both an `id` and `description`

Not that we should, but I believe that just an eslint rule that we've enabled. (Rightfully, as it's recommended by formatjs, based on general experiences of managing strings) I believe we can opt out of it in specific cases, should we choose, with no effect.

```js
intl.formatMessage({
  // eslint-disable-next-line formatjs/no-id
  id: "foo",
  defaultMessage: "Work preferences",
  description: "Title for Profile Form wrapper  in Work Preferences Form",
})
```

I didn't realize when I created the issues, but now I'm not sure this is possible with how auto-generation of IDs works. I didn't realize this suggestion meant ignoring that eslint rule 😄
